controlplane/pkg/plugins/utils.go

```diff
@@ -71,17 +83,25 @@ func createPlugin(name, endpoint string, services map[plugins.PluginCapability]i
 	return nil
 }
 
-func registerPlugin(name, endpoint, registry string, capabilities []plugins.PluginCapability) error {
+func (rm *registrationMonitor) registerPlugin(name, endpoint, registry string, capabilities []plugins.PluginCapability) error {
 	_ = tools.WaitForPortAvailable(context.Background(), "unix", registry, 100*time.Millisecond)
 	conn, err := tools.DialUnix(registry)
 	if err != nil {
 		return fmt.Errorf("cannot connect to the Plugin Registry: %v", err)
 	}
-	defer func() {
+
+	cleanup := func() {
```

Comment:
Should we close `conn` in case of `livenessClient` tracks that `pluginRegistry` is down?